content/webapp/services/iiif/fetch/manifest.ts

```diff
@@ -1,32 +1,37 @@
 import { Manifest } from '@iiif/presentation-3';
 
 async function getIIIFManifest(url: string): Promise<Manifest | undefined> {
-  const resp = await fetch(url);
+  try {
+    const resp = await fetch(url);
 
-  if (resp.status === 404) {
-    const bnumber = url.match(/b[0-9]{7}[0-9x]/)?.[0];
+    if (resp.status === 404) {
+      const bnumber = url.match(/b[0-9]{7}[0-9x]/)?.[0];
 
-    // We have to add this condition because of a known issue with BD works and Archivemetica
-    // We don't want the error to be sent if it's part of the known issue.
-    // https://github.com/wellcomecollection/wellcomecollection.org/issues/10907
-    // TODO: Check in once in a while to see if it's been fixed. There is not ticket to link to currently for that piece of work.
+      // We have to add this condition because of a known issue with BD works and Archivemetica
+      // We don't want the error to be sent if it's part of the known issue.
+      // https://github.com/wellcomecollection/wellcomecollection.org/issues/10907
+      // TODO: Check in once in a while to see if it's been fixed. There is not ticket to link to currently for that piece of work.
 
-    // TEMP FIX: If location doesn't contain a B number, we assume that it is a Born Digital work.
-    // This works for now but won't in the future as more BD works come in
-    // TODO: Change how it's done once https://github.com/wellcomecollection/catalogue-pipeline/issues/2659 is through.
-    if (bnumber) {
-      const dashboardUrl = `https://iiif.wellcomecollection.org/dash/Manifestation/${bnumber}`;
+      // TEMP FIX: If location doesn't contain a B number, we assume that it is a Born Digital work.
+      // This works for now but won't in the future as more BD works come in
+      // TODO: Change how it's done once https://github.com/wellcomecollection/catalogue-pipeline/issues/2659 is through.
+      if (bnumber) {
+        const dashboardUrl = `https://iiif.wellcomecollection.org/dash/Manifestation/${bnumber}`;
 
-      throw new Error(
-        `Tried to retrieve IIIF Manifest at ${url}, but it's 404-ing. To fix, try running the "Rebuild IIIF" task in the iiif-builder dashboard. ${dashboardUrl}`
-      );
+        throw new Error(
+          `Tried to retrieve IIIF Manifest at ${url}, but it's 404-ing. To fix, try running the "Rebuild IIIF" task in the iiif-builder dashboard. ${dashboardUrl}`
+        );
+      }
+
+      return undefined;
     }
 
-    return undefined;
+    const manifest = await resp.json();
+    return manifest;
+  } catch (e) {
+    console.error(`Something went wrong fetching ${url}`);
```

Comment:
would it make sense to log the `e.message` here as well?

Reply:
Probably yes! I quickly tried to find out if we had a set format for errors, but I think not really? We could make one if not? We do it for [`appError`](https://github.com/wellcomecollection/wellcomecollection.org/blob/trycatch-iiifmanifest/common/services/app/index.ts#L9)and [`wellcomeAPIerror`](https://github.com/wellcomecollection/wellcomecollection.org/blob/trycatch-iiifmanifest/content/webapp/services/wellcome/index.ts#L112), but they're not exactly what we'd like for this.

Reply:
I've added [this](https://github.com/wellcomecollection/wellcomecollection.org/pull/11064/files#diff-ba740c986a21caf40f96e44870e45a6f6b3130763c0b2cad510d26ce864e2e53R34) for now!